src/test/ui/consts/const-labeled-break.rs

```diff
@@ -4,7 +4,6 @@
 // See https://github.com/rust-lang/rust/issues/51350 for more information.
 
 const CRASH: () = 'a: while break 'a {};
+//~^ ERROR constant contains unimplemented expression type
```

Comment:
ugh, that's a breaking change.

@eddyb what if we did const prop and branch cleanup before const qualif?

Actually, just branch cleanup may suffice here. @Centril 	what's the MIR of this constant after your changes?

Reply:
The issue is probably that we now generate a `FalseUnwind` for while loops now. We could ignore them in const qualification and allow all "loops that can't ever loop" in consts but they have to stay in the MIR until borrow checking.

Reply:
> ugh, that's a breaking change.

That's... debatable. :)

From my perspective this is a bug-fix that fixes the unjustified semantic divergence between `loop`- and `while`-expressions. Moreover, before 1.33.0 this was an ICE or error all the way back to 1.0.0. Also, you have to literally write `'label: while break 'label {}`. Other forms you might think should behave similarly such as `while break {}` do not work. I also find it highly unlikely that anyone would ever write `'label: while break 'label {}` because it is useless.

I r+ed this test in https://github.com/rust-lang/rust/pull/60360 because it documented the behavior of the compiler.  But now that the behavior is fixed so should the test.

> @eddyb what if we did const prop and branch cleanup before const qualif?

That would make const-prop a part of our stability guarantees. I think that's a non-starter from a T-Lang POV.

> We could ignore them in const qualification and allow all "loops that can't ever loop" in consts but they have to stay in the MIR until borrow checking.

We could... but that would allow more behavior that is disallowed now. Also, I think it is not worth any of y'all's time to make `'label: while break 'label {}` continue to work.

Reply:
Wait, so what's the current behavior, *ignoring constants*, for `'label: while break 'label {}`?
Is it a noop, or an error?

Reply:
A noop, see https://discordapp.com/channels/442252698964721669/443151243398086667/593054967624105997.

Reply:
@matthewjasper I looked into `FalseUnwind` and indeed it should've been ignored in const-checking, see #62272.